### ISSUE TITLE
Op [broadcastdiv] X argument uses leaked workspace pointer from workspace

### ISSUE THREAD

**Issue by Yu-Hang** [deeplearning4j/deeplearning4j#5412]
Hi,

I'm trying to write the output of each layer for debugging purpose. 

Here is my code:

```
// test to get output from each layer------------------------------------------------------------------
        ImageRecordReader recordReader_test = new ImageRecordReader(height, width, channels, labelMaker);
        try {
            recordReader_test.initialize(test);
        } catch (IOException e) {
            e.printStackTrace();
        }
        recordReader_test.setListeners(new LogRecordListener());
        DataSetIterator dataIter_test = new RecordReaderDataSetIterator(recordReader_test, batchSize, 1, numLabels);
        scaler.fit(dataIter_test);
        dataIter_test.setPreProcessor(scaler);

        DataSet testDataSet =  dataIter_test.next();

        List<Pair<INDArray, INDArray>> embeddingByEpoch = new ArrayList<>();
        List<Integer> epochNum = new ArrayList<>();


        PrintWriter writer_test = new PrintWriter("log_model2_testOn_5_layer_output.txt", "UTF-8");
        for (int i=0;i < epochs; i++) {
            model.fit(dataIter);

            log.info("*** Completed epoch {} ***", i);

            //Feed forward to the embedding layer (layer 5) to get the 2d embedding to plot later
            INDArray embedding = model.feedForwardToLayer(2, testDataSet.getFeatures()).get(3);
            INDArray embedding1 = model.feedForwardToLayer(2, testDataSet.getFeatures()).get(4);
            INDArray embedding2 = model.feedForwardToLayer(2, testDataSet.getFeatures()).get(5);
            INDArray embedding3 = model.feedForwardToLayer(2, testDataSet.getFeatures()).get(6);

            embeddingByEpoch.add(new Pair<>(embedding, testDataSet.getLabels()));
            epochNum.add(i);

            writer_test.println(embedding);
            writer_test.println("\n");
            writer_test.println(embedding1);
            writer_test.println("\n");
            writer_test.println(embedding2);
            writer_test.println("\n");
            writer_test.println(embedding3);
        }

        writer_test.close();
```

However, I got error:

```
May 31, 2018 5:52:12 PM org.dp.importK.malariaModel run
INFO: *** Completed epoch 0 ***
Exception in thread "main" org.nd4j.linalg.exception.ND4JIllegalStateException: Op [broadcastdiv] X argument uses leaked workspace pointer from workspace [LOOP_EXTERNAL]
```

Thanks,

**Comment by raver119** [comment]
1) Always provide full stack trace, not just 1 line.
2) in your case stuff you're doing is unsafe due to workspaces. Easies workaround: disable workspaces  for inference, and use appropriate signature for FF. Or just disable both workspace modes.

**Comment by Yu-Hang** [comment]
Exception in thread "main" org.nd4j.linalg.exception.ND4JIllegalStateException: Op [broadcastdiv] X argument uses leaked workspace pointer from workspace [LOOP_EXTERNAL]
For more details, see the ND4J User Guide: nd4j.org/userguide#workspaces-panic
	at org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner.checkForWorkspaces(DefaultOpExecutioner.java:436)
	at org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner.profilingHookIn(DefaultOpExecutioner.java:490)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaExecutioner.exec(CudaExecutioner.java:106)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaGridExecutioner.pushToGrid(CudaGridExecutioner.java:253)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaGridExecutioner.processAsGridOp(CudaGridExecutioner.java:346)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaGridExecutioner.invoke(CudaGridExecutioner.java:713)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaGridExecutioner.exec(CudaGridExecutioner.java:114)
	at org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner.execAndReturn(DefaultOpExecutioner.java:253)
	at org.deeplearning4j.nn.layers.normalization.BatchNormalization.preOutput(BatchNormalization.java:335)
	at org.deeplearning4j.nn.layers.normalization.BatchNormalization.activate(BatchNormalization.java:226)
	at org.deeplearning4j.nn.layers.AbstractLayer.activate(AbstractLayer.java:310)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.activationFromPrevLayer(MultiLayerNetwork.java:815)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.feedForwardToLayer(MultiLayerNetwork.java:987)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.feedForwardToLayer(MultiLayerNetwork.java:952)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.feedForwardToLayer(MultiLayerNetwork.java:939)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.feedForwardToLayer(MultiLayerNetwork.java:924)
	at org.dp.importK.malariaModel.run(malariaModel.java:162)
	at org.dp.importK.malariaModel.main(malariaModel.java:60)

**Comment by raver119** [comment]
Thanks, i see.

Use proposed solutions please, and you'll be fine.
